src/main/java/net/mcreator/generator/setup/WorkspaceGeneratorSetup.java

```diff
@@ -81,17 +83,9 @@ public static void cleanupGeneratorForSwitchTo(Workspace workspace, GeneratorCon
 		moveFilesToAnotherDir(workspace.getFolderManager().getSoundsDir(),
 				GeneratorUtils.getSpecificRoot(workspace, newGenerator, "sounds_dir"));
 
-		moveFilesToAnotherDir(workspace.getFolderManager().getBlocksTexturesDir(),
-				GeneratorUtils.getSpecificRoot(workspace, newGenerator, "block_textures_dir"));
-
-		moveFilesToAnotherDir(workspace.getFolderManager().getItemsTexturesDir(),
-				GeneratorUtils.getSpecificRoot(workspace, newGenerator, "item_textures_dir"));
-
-		moveFilesToAnotherDir(workspace.getFolderManager().getArmorTexturesDir(),
-				GeneratorUtils.getSpecificRoot(workspace, newGenerator, "armor_textures_dir"));
-
-		moveFilesToAnotherDir(workspace.getFolderManager().getOtherTexturesDir(),
-				GeneratorUtils.getSpecificRoot(workspace, newGenerator, "other_textures_dir"));
+		Arrays.stream(TextureType.values()).forEach(category -> moveFilesToAnotherDir(
+				workspace.getFolderManager().getTexturesFolder(category),
+				GeneratorUtils.getSpecificRoot(workspace, newGenerator, category + "_textures_dir")));

```

Comment:
Shouldn't this be category.getID(), so it is lowercasE?

Reply:
When I tested with 1.16.5 and 1.17.1 (before switching to 1.18.2). I re-tested and it also seemed to work, but I change it, so no matter what, it will work.